libfreerdp/gdi/gdi.c

```diff
@@ -641,18 +676,36 @@ static void gdi_patblt(rdpContext* context, PATBLT_ORDER* patblt)
 		else
 		{
 			data = (BYTE*) _aligned_malloc(8 * 8 * gdi->bytesPerPixel, 16);
+			if (!data)
+			{
+				ret = FALSE;
+				goto out_error;
+			}
 
 			freerdp_image_copy_from_monochrome(data, gdi->format, -1, 0, 0, 8, 8,
 					brush->data, backColor, foreColor, gdi->palette);
 		}
 
 		hBmp = gdi_CreateBitmap(8, 8, gdi->drawing->hdc->bitsPerPixel, data);
+		if (!hBmp)
+		{
+			_aligned_free(data);
+			ret = FALSE;
+			goto out_error;
+		}
 
 		originalBrush = gdi->drawing->hdc->brush;
 		gdi->drawing->hdc->brush = gdi_CreatePatternBrush(hBmp);
+		if (!gdi->drawing->hdc->brush)
+		{
+			_aligned_free(data);
+			ret = FALSE;
+			goto out_error;
+		}
 
-		gdi_PatBlt(gdi->drawing->hdc, patblt->nLeftRect, patblt->nTopRect,
-				patblt->nWidth, patblt->nHeight, gdi_rop3_code(patblt->bRop));
+		if (gdi_PatBlt(gdi->drawing->hdc, patblt->nLeftRect, patblt->nTopRect,
+				patblt->nWidth, patblt->nHeight, gdi_rop3_code(patblt->bRop)) != 1)
```

Comment:
This is ok for freerdp's api but I'd suggest to check for == 0 to match the msdn description.